02-express-tutorial/answers/controllers/people.js

```diff
@@ -0,0 +1,58 @@
+let { people } = require('../../data');
+
+const getPeople = (req, res) => {
+	res.status(200).json({ success: true, data: people });
+};
+
+const createPerson = (req, res) => {
+	const { name } = req.body;
+	!name
+		? res.status(400).json({ success: false, msg: 'please provide name value' })
+		: null;
+	res.status(201).json({ success: true, person: name });
+};
+
+const createPersonPostman = (req, res) => {
+	const { name } = req.body;
+	!name
+		? res.status(400).json({ success: false, msg: 'please provide name value' })
+		: null;
+	res.status(201).json({ success: true, data: [...people, name] });
+};
+
+const updatePerson = (req, res) => {
+	const { id } = req.params;
+	const { name } = req.body;
+	const person = people.find((person) => person.id === Number(id));
+	!person
+		? res.status(404).json({ success: false, msg: `no person with id ${id}` })
+		: null;
+	const newPeople = people.map((person) => {
+		if (person.id === Number(id)) {
+			person.name = name;
+		}
+		return person;
+	});
+	res.status(200).json({ success: true, data: newPeople });
+};
+
+const deletePerson = (req, res) => {
+	const person = people.find((person) => person.id === Number(req.params.id));
+	!person
+		? res
+				.status(404)
+				.json({ success: false, msg: `no person with id ${req.params.id}` })
+		: null;
```

Comment:
Check if this needs to be rewritten to avoid node server erorrs from being emmited

Reply:
Yes, correction made. Thank you!